synapse/storage/databases/main/events_worker.py

```diff
@@ -738,13 +748,29 @@ def _get_events_from_cache(
         event_map = {}
 
         for event_id in events:
+            # First check if its in the event cache
             ret = self._get_event_cache.get(
                 (event_id,), None, update_metrics=update_metrics
             )
-            if not ret:
-                continue
+            if ret:
+                event_map[event_id] = ret
+
+            # Otherwise check if we still have the event in memory.
```

Comment:
"Otherwise" is confusing here. We run through all of lines 758--771 even if there was a (strong) cache hit on lines 752--755. Isn't that redundant if `ret` was truthy?

Do we want a `continue` on 757 or similar bit of control flow?

Reply:
Oh, yes, you're quite right.